confd/service/command.go

```diff
@@ -0,0 +1,52 @@
+package service
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+func executeCommand(command string) error {
+	cmd := exec.Command("/bin/sh", "-c", command)
+	err := cmd.Start()
+	if err != nil {
+		return errors.Wrapf(err, "failed to execute command: \"%s\"", command)
+	}
+
+	return cmd.Wait()
+}
+
+func post(command string) error {
+	log.Println("execute post command", command)
+	err := executeCommand(command)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute post command")
+	}
+	return nil
+}
+
+func preCheck(conf Configuration, data interface{}) error {
+	dir := filepath.Dir(conf.Target)
+	prefix := filepath.Base(conf.Target)
+	tmpFile, err := ioutil.TempFile(dir, prefix)
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp file for pre check")
+	}
+
+	defer os.Remove(tmpFile.Name())
```

Comment:
**[checkstyle]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
error return value not checked (defer os.Remove(tmpFile.Name()))